cmd/prowjob/healthCheck.go

```diff
@@ -0,0 +1,197 @@
+)
+
+// HealthCheckConfig represents configuration of external services which status will be checked
+type HealthCheckConfig struct {
+	ExternalServices []Service `json:"externalServices"`
+}
+
+// HealthCheckStatus contains specification and current status of external services
+// and contains a map of currently unhealthy critical components
+type HealthCheckStatus struct {
+	ExternalServices            []Service           `json:"externalServices"`
+	UnhealthyCriticalComponents map[string][]string `json:"unhealthyCriticalComponents"`
+}
+
+// Service contains specification of an external service specified via config
+// and holds an information about current status of related components
+type Service struct {
+	Name               string         `json:"name"`
+	CriticalComponents []string       `json:"criticalComponents"`
+	StatusPageURL      string         `json:"statusPageURL"`
+	CurrentStatus      status.Summary `json:"currentStatus"`
+}
+
+// healthCheckCmd represents the createReport command
+var healthCheckCmd = &cobra.Command{
+	Use:   "health-check",
+	Short: `Perform a health check on dependant services`,
+	Long:  healthCheckCmdLongDescription,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		if viper.ConfigFileUsed() == "" {
+			viper.SetConfigFile(healthCheckDefaultConfigPath)
+		}
+		if err := viper.ReadInConfig(); err != nil {
+			return fmt.Errorf("err readinconfig: %+v", err)
+		}
+		if err := viper.Unmarshal(&healthCheckConfig); err != nil {
+			return fmt.Errorf("failed to parse config: %+v", err)
+		}
+		if viper.GetBool(notifyOnPRParamName) {
+			for _, e := range healthCheckNotifyRequiredEnvVars {
+				if viper.GetString(e) == "" {
+					return fmt.Errorf("%q flag provided, but %q env var not set", notifyOnPRParamName, strings.ToUpper(e))
+				}
+			}
+		}
+		return nil
+	},
+	SilenceUsage: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		hcStatus := &HealthCheckStatus{}
+		hcStatus.ExternalServices = healthCheckConfig.ExternalServices
+		hcStatus.UnhealthyCriticalComponents = make(map[string][]string)
+
+		for i, service := range hcStatus.ExternalServices {
+			r, err := http.Get(service.StatusPageURL)
+			if err != nil {
+				return fmt.Errorf("failed to get service %s status page: %+v", service.Name, err)
+			}
+			body, err := io.ReadAll(r.Body)
+			if err != nil {
+				return fmt.Errorf("failed to read response body for a service %s: %+v", service.Name, err)
+			}
+			v := status.Summary{}
+			if err := json.Unmarshal(body, &v); err != nil {
+				return fmt.Errorf("failed to unmarshal response body from a service %s: %+v", service.Name, err)
+			}
+			hcStatus.ExternalServices[i].CurrentStatus = v
+
+			for _, c := range v.Components {
+				if c.Status == "major_outage" && isCriticalComponent(service, c) {
+					hcStatus.UnhealthyCriticalComponents[service.Name] = append(hcStatus.UnhealthyCriticalComponents[service.Name], c.Name)
+				}
```

Comment:
if there's currently no outage, it is possible to test the PR notification by adding e.g. this code under this line
```
				if service.Name == "quay" && c.Name == "API" {
					hcStatus.UnhealthyCriticalComponents[service.Name] = append(hcStatus.UnhealthyCriticalComponents[service.Name], c.Name)
				}
```